src/utils/scales-utils.js

```diff
@@ -150,6 +150,32 @@ export function _getSmallestDistanceIndex(values, scaleObject) {
   return result;
 }
 
+/**
+ * This is a workaround for issue that ordinal scale
+ * does not have invert method implemented in d3-scale.
+ * @param {Object} Ordinal d3-scale object.
+ * @returns {void}
+ * @private
+ */
+
+function addInvertFunctionToOrdinalScaleObject(scale) {
+  if (scale.invert) return;
+
+  scale.invert = function invert(value) {
+    let domainIndex = 0;
+    const n = scale.domain().length;
+    const reverse = scale.range()[1] < scale.range()[0];
```

Comment:
maybe cleaner to pull the start/end values out rather than to keep calling range()

Reply:
Not completely sure what do you mean here; we need to figure out whether range is ascending or descending in order to see what is 'start' (lower range) and 'end' (higher range).

Reply:
you call scale.range() fourish times, it might be better to do something like:
```js
const [lower, upper] = scale.range();
const start = Math.min(lower, upper);
const end = Math.max(lower, upper);
```
That way there is less index manipulation

Reply:
👍